From 6a3567780adb01e483332e576138b7ae0a32265d Mon Sep 17 00:00:00 2001
From: Yong Gan <yong.gan@nxp.com>
Date: Wed, 10 Oct 2018 08:37:57 +0800
Subject: [PATCH 4780/5242] MGS-4285 [#imx-1211] gl compositor stress test
 failed with out of memory

commit  25a634a94e7ed21447da676b7ba0e97d8026ed2f from
https://source.codeaurora.org/external/imx/linux-imx.git

No need to add extra ref count for the dma fence in the API viv_fence_create.
Incrence the ref-count in dma_fence_init and sync_file_create
decrease the ref-count in gckOS_Signal and close(fd).

Signed-off-by Yong Gan <yong.gan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../hal/os/linux/kernel/gc_hal_kernel_sync.c       |    1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_sync.c b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_sync.c
index 5cda578..37c95ab 100644
--- a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_sync.c
+++ b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_sync.c
@@ -349,7 +349,6 @@ struct dma_fence * viv_fence_create(struct viv_sync_timeline *timeline,
 
     if (!signal->done) {
         signal->fence = (struct dma_fence*)fence;
-        dma_fence_get((struct dma_fence*)fence);
     }
 
     spin_unlock(&signal->lock);
-- 
1.7.9.5

